src/components/hooks/usePeoplePage.ts

```diff
@@ -0,0 +1,99 @@
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { getPeople } from '../../api/data';
+import debounce from '../../helpers/debounce';
+
+export const usePeoplePage = () => {
+  const history = useHistory();
+  const location = useLocation();
```

Comment:
I found the next information:

```
"Location
The location object represents where the app is now, where you want it to go, or even where it was.

It’s also found on history.location but you shouldn’t use that because it’s mutable.

A location object is never mutated so you can use it in the lifecycle hooks to determine when navigation happens. This is really useful for data fetching or DOM side effects."
```

https://www.freecodecamp.org/news/hitchhikers-guide-to-react-router-v4-4b12e369d10/

so, what should I do in my case?